Src/WindowsServer/WindowsServer.Shared.Tests/FirstChanceExceptionStatisticsTelemetryModuleTest.cs

```diff
@@ -0,0 +1,204 @@
+    {
+        private TelemetryConfiguration configuration;
+        private IList<ITelemetry> items;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            this.items = new List<ITelemetry>();
+
+            this.configuration = new TelemetryConfiguration();
+
+            this.configuration.TelemetryChannel = new StubTelemetryChannel
+            {
+                OnSend = telemetry => this.items.Add(telemetry),
+                EndpointAddress = "http://test.com"
+            };
+            this.configuration.InstrumentationKey = "MyKey";
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            this.configuration = null;
+            this.items.Clear();
+        }
+
+        [TestMethod]
+        public void FirstChanceExceptionStatisticsTelemetryModuleDoNotThrowOnNullException()
+        {
+            EventHandler<FirstChanceExceptionEventArgs> handler = null;
+            using (var module = new FirstChanceExceptionStatisticsTelemetryModule(
+                h => handler = h,
+                _ => { }))
+            {
+                module.Initialize(this.configuration);
+                handler.Invoke(null, new FirstChanceExceptionEventArgs(null));
+            }
+        }
+
+        [TestMethod]
+        public void FirstChanceExceptionStatisticsTelemetryModuleDoNotCauseStackOverflow()
+        {
+            this.configuration.TelemetryInitializers.Add(new StubTelemetryInitializer()
+            {
+                OnInitialize = (item) =>
+                {
+                    throw new Exception("this exception may cause stack overflow");
+                }
+            });
+
+            using (var module = new FirstChanceExceptionStatisticsTelemetryModule())
+            {
+                module.Initialize(this.configuration);
+
+                try
+                {
+                    throw new Exception("test");
+                }
+                catch (Exception exc)
+                {
+                    Assert.Equal("test", exc.Message);
+                }
+            }
+        }
+
+        [TestMethod]
+        public void FirstChanceExceptionStatisticsTelemetryModuleTracksMetricWithTypeAndMethodOnException()
+        {
+            var metrics = new List<KeyValuePair<Metric, double>>();
+            this.configuration.MetricProcessors.Add(new StubMetricProcessor()
+            {
+                OnTrack = (m, v) =>
+                {
+                    metrics.Add(new KeyValuePair<Metric, double>(m, v));
+                }
+            });
+
+            using (var module = new FirstChanceExceptionStatisticsTelemetryModule())
+            {
+                module.Initialize(this.configuration);
+
+                try
+                {
+                    throw new Exception("test");
+                }
+                catch (Exception exc)
+                {
+                    Assert.Equal("test", exc.Message);
```

Comment:
If this assert were further down and testing that the module did something then it would have value.